tidal/content/contents/code/tidal.js

```diff
@@ -129,76 +107,97 @@ var TidalResolver = Tomahawk.extend( TomahawkResolver, {
     },
 
     _convertArtist: function (entry) {
+        Tomahawk.log('Called _convertArtist()');
         return entry.name;
     },
 
     search: function (qid, query) {
-        if (!this._countryCode) return;
+        Tomahawk.log('Called search()');
+        if (!this.logged_in) this._login(this.search, [qid, query], this);
+
         var that = this;
 
-        query = encodeURIComponent(query);
+        var query = encodeURIComponent(query);
 
-        xhrRequest(that.api_location + "search/tracks","GET", "limit=10000&query=" 
-                +query+"&token="+that.api_token+"&countryCode="+that._countryCode,
-            function (request) {
-                var resp = JSON.parse(request.responseText);
-                var data = {
-                    qid: qid,
-                    results: resp.items.map(that._convertTrack),
-                }
-                Tomahawk.addTrackResults(data);
+        var params = {
+            limit: 25,
+            query: query ,
+            token: this.api_token ,
+            countryCode: this._countryCode
+        };
+
+        return Tomahawk.ajax(this.api_location + "search/tracks",
+            {
+                data: params
             }
-        );
-        xhrRequest(that.api_location + "search/albums","GET", "limit=10000&query=" 
-                +query+"&token="+that.api_token+"&countryCode="+that._countryCode,
-            function (request) {
-                var resp = JSON.parse(request.responseText);
-                var data = {
-                    qid: qid,
-                    albums: resp.items.map(that._convertAlbum),
-                }
-                Tomahawk.addAlbumResults(data);
+        ).then( function (response) {
+            var data = {
+                qid: qid,
```

Comment:
I see, 'Called search()' never got printed, so I haven't even had a chance to debug this. I just replaced the xhr stuff like for like.